tests/net/lib/mqtt_packet/src/mqtt_packet.c

```diff
@@ -513,6 +532,18 @@ static ZTEST_DMEM
 u8_t unsuback1[] = {0xb0, 0x02, 0x00, 0x01};
 static ZTEST_DMEM struct mqtt_unsuback_param msg_unsuback1 = {.message_id = 1};
 
+static ZTEST_DMEM
+u8_t max_pkt_len[] = {0x30, 0xff, 0xff, 0xff, 0x7f};
```

Comment:
I'm confused. Are we reading four or five bytes here, if this was the test case that passes?

Reply:
Here we read four bytes, holding the maximum length allowed by the standard.

I've added two test cases, one that parses maximum allowed length (the one you've pointed), the other that tries to encode length on 5 bytes. The latter is expected to return an error in order to pass the test.

Reply:
Ok, I was just unsure what the five bytes of max_pkt_len[] represented, since the maximum length is four.

Reply:
The function which is verified parses an entire MQTT fixed header, which is 1 byte for packet type and flags, and up to 4 bytes for length. Hence 5 bytes in total.

Reply:
Good, now it is clear.